src/scipp/interpolate/_interp1d.py

```diff
@@ -0,0 +1,99 @@
+
+    :param kind: Specifies the kind of interpolation as a string or as an integer
+                 specifying the order of the spline interpolator to use. The string
+                 has to be one of 'linear', 'nearest', 'nearest-up', 'zero', 'slinear',
+                 'quadratic', 'cubic', 'previous', or 'next'. 'zero', 'slinear',
+                 'quadratic' and 'cubic' refer to a spline interpolation of zeroth,
+                 first, second or third order; 'previous' and 'next' simply return the
+                 previous or next value of the point; 'nearest-up' and 'nearest' differ
+                 when interpolating half-integers (e.g. 0.5, 1.5) in that 'nearest-up'
+                 rounds up and 'nearest' rounds down. Default is 'linear'.
+    :param fill_value: Set to 'extrapolate' to allow for extrapolation of points
+                       outside the range.
+
+    Examples:
+
+      >>> x = sc.geomspace(dim='x', start=0.1, stop=0.4, num=4, unit='rad')
+      >>> da = sc.DataArray(sc.sin(x), coords={'x': x})
+
+      >>> from scipp.interpolate import interp1d
+      >>> f = interp1d(da, 'x')
+
+      >>> xnew = sc.linspace(dim='x', start=0.1, stop=0.4, num=5, unit='rad')
+      >>> f(xnew)  # use interpolation function returned by `interp1d`
+      <scipp.DataArray>
+      Dimensions: Sizes[x:5, ]
+      Coordinates:
+        x                         float64            [rad]  (x)  [0.100000, 0.175000, 0.250000, 0.325000, 0.400000]
+      Data:
+                                  float64  [dimensionless]  (x)  [0.099833, 0.173987, 0.247384, 0.318433, 0.389418]
+
+      >>> f(xnew, midpoints=True)
+      <scipp.DataArray>
+      Dimensions: Sizes[x:4, ]
+      Coordinates:
+        x                         float64            [rad]  (x [bin-edge])  [0.100000, 0.175000, 0.250000, 0.325000, 0.400000]
+      Data:
+                                  float64  [dimensionless]  (x)  [0.137015, 0.210685, 0.282941, 0.353926]
+    """  # noqa #501
+    import scipy.interpolate as inter
+
+    def func(xnew: Variable, *, midpoints=False) -> DataArray:
+        """Compute interpolation function defined by ``interp1d`` at interpolation points.
+
+        :param xnew: Interpolation points.
+        :param midpoints: Interpolate at midpoints of given points. The result will be
+                          a histogram. Default is ``False``.
+        :return: Interpolated data array with new coord given by interpolation points
+                 and data given by interpolation function evaluated at the
+                 interpolation points (or evaluated at the midpoints of the given
+                 points).
+        """
+        if xnew.unit != da.coords[dim].unit:
+            raise UnitError(
+                f"Unit of interpolation points '{xnew.unit}' does not match unit "
+                f"'{da.coords[dim].unit}' of points defining the interpolation "
+                "function.")
```

Comment:
Naming the dim / coord would be nice. Something like
```suggestion
                f"function in dimension {dim}.")
```